Step2 (JEJE)

jung-yun opened the pull request:
Step2 구현하였습니다.

CardDeck structure를 만들고 test scenario를 view controller에 추가했습니다.

jung-yun committed:
Replace guard let with nil coalescing operator

jung-yun committed:
Add CardDeck structure

Add member property and methods as well

jung-yun committed:
Add test scenario

godrm commented on PokerGameApp/PokerGameApp/CardDeck.swift:
```diff
@@ -0,0 +1,47 @@
+    private func generateCardSet() -> Array<String> {
+        var result: Array<String> = []
+        let suitArray: Array<String> = ["♠️","♥️","♦️","☘️"]
+        let valueArray: Array<String> = ["1","2","3","4","5","6","7","8","9","10","J","Q","K"]
```
이 단계 요구사항이 CardDeck인가요? Card 설계하는 게 빠진 것 같은데요?

jung-yun replied:
step2를 하지 않고 바로 step3 보고있었네요... 죄송합니다 수정했습니다~!🙏

godrm commented:
일단 PR 열어 놓을테니까 확인해보고 수정해서 Push하세요.

jung-yun committed:
Add Card class

Implemented card class using nested enum
Edited relevant code in other classes (CardDeck, ViewController) - Now currentCardDeck is made of Card instances, and removeOne method returns optional Card instance.

jung-yun committed:
Add comment

jung-yun committed:
Use unicode in String literal instead of emoji

jung-yun committed:
Edit unicode in string literal

godrm commented on PokerGameApp/PokerGameApp/Card.swift:
```diff
@@ -0,0 +1,42 @@
+        case spades = "\u{2660}"
+        case hearts = "\u{2665}"
+        case diamonds = "\u{2666}"
+        case clovers = "\u{2618}"
```
유니코드는 "♥️"를 그대로 표기해도 됩니다. 더 읽기 편합니다.

godrm commented on PokerGameApp/PokerGameApp/Card.swift:
```diff
@@ -0,0 +1,42 @@
+    }
+   
+    init(suit: Suit, value: Value) {
+        self.cardDescription = "\(suit.rawValue)\(value.rawValue)"
```
rawValue에 접근하지 말고 Suit나 Value에도 CustomStringConvertible 프로토콜을 채택해보세요.

godrm commented:
어쩐지 이상하다 했습니다. 다음 단계를 하면서 개선해보세요.
